### DISCUSSION TITLE
Trying to Execute PowerShell script with Wix v4

### DISCUSSION THREAD

**Discussion by mikitheking5** [wixtoolset/issues#7713] [Questions]
The following code returns error  code 1603, but I have no idea what could be the problem, the documentation says this is how this feature is used, what Am i missing here, is this a bug or what? I would very much appreciate somebody's assistance, I'm new to this but this is really interesting to me...

 ```
<Component Id="djole.ps1" Guid="" DiskId="1" Directory="$(var.ConsoleApp1.ProjectDir)">
  <File Id="djole.ps1" Name="djole.ps1" Source="djole.ps1"/>
 </Component>

 <Property Id="POWERSHELLEXE">
  <RegistrySearch Id="POWERSHELLEXE"
	  Type="raw"
	  Root="HKLM"
	  Key="SOFTWARE\Microsoft\PowerShell\1\ShellIds\Microsoft.PowerShell"
	  Name="Path" />
 </Property>
 

  
<SetProperty
      Id="WixSilentExecCmdLine"
Value="&quot;[POWERSHELLEXE]&quot; -Version 5.1.19041.2673 -NoProfile -NonInteractive -InputFormat None -ExecutionPolicy Bypass -Command &quot;&amp; '[#djole.ps1]' ; exit $$($Error.Count)&quot;" 
Before="ConvolutedCoreCountCustomAction"
Sequence="execute"
/>
<CustomAction
	Id="ConvolutedCoreCountCustomAction"
	BinaryRef="Wix4UtilCA_$(sys.BUILDARCHSHORT)"
	DllEntry="WixSilentExec"
	Execute="immediate"
	Return="check"
/>

<InstallExecuteSequence>
 <Custom Action="ConvolutedCoreCountCustomAction" Before="InstallInitialize" />
</InstallExecuteSequence>
```

Also, I get this ""WIX0014	The Component/@Directory attribute's value, 'C:\Users\...\Desktop\Bild\ConsoleApp1\ConsoleApp1\', is not a legal identifier. Identifiers may contain ASCII characters A-Z, a-z, digits, underscores (_), or periods (.). Every identifier must begin with either a letter or an underscore.	ConsoleApp1.installer""

Why do I get this error when I never used the illegal characters?

**Comment by barnson** [comment]
WiX doesn't use issues for support. We moved this issue to the [GitHub discussions area][gh]. You might also consider asking your question on [Stack Overflow][so].

[gh]: https://github.com/wixtoolset/issues/discussions?discussions_q=category%3AQuestions
[so]: https://stackoverflow.com/questions/tagged/wix

**Reply by mikitheking5** [reply]
Thanks for that @barnson. I Just noticed there is a powershell extension as a NuGet package but there's no documentation on how to use it and now examples/sample files. If somebody could help me with this or point me to the right place I'd very much appreciate it.
